London | May-2025 | Mohamed Ibrahim | Sprint-1

sidicamoli opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Sprint-1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

- Commented out instruction lines in `Sprint-1/2-mandatory-errors/0.js` as requested

## Questions

None for now, thanks!

cjyuan commented on Sprint-1/1-key-exercises/1-count.js:
```diff
@@ -4,3 +4,5 @@ count = count + 1;
 
 // Line 1 is a variable declaration, creating the count variable with an initial value of 0
 // Describe what line 3 is doing, in particular focus on what = is doing
+// Line 3 takes the current value of count (0), adds 1 to it, and assigns the result (1) back to count.
+// The = operator is used for assignment — it sets the variable to a new value.
```
Operation like `count = count + 1` is very common in programming, and there is a programming term describing such operation.

Can you find out what one-word programming term describes the operation on line 3?

cjyuan commented on Sprint-1/1-key-exercises/4-random.js:
```diff
@@ -4,6 +4,30 @@ const maximum = 100;
+A; Num will be a random integer between 1 and 100, inclusive — meaning it can be any whole number from 1 to 100.
 // Try breaking down the expression and using documentation to explain what it means
+A; Breakdown of the expression:
+1. Math.random()
+Returns a random decimal number between 0 (inclusive) and 1 (exclusive).
```
We can also use the concise and precise ***interval notation*** to describe a range of values.
For example, we can say, "`Math.random()` returns a random number in the interval `[0, 1)`"

cjyuan commented on Sprint-1/3-mandatory-interpret/3-to-pounds.js:
```diff
@@ -25,3 +25,14 @@ console.log(`£${pounds}.${pence}`);
+// 2. const penceStringWithoutTrailingP = penceString.substring(0, penceString.length - 1); Removes the trailing "p" character from the string by slicing off the last character.
+// 3.const paddedPenceNumberString = penceStringWithoutTrailingP.padStart(3, "0"); Ensures the number is at least 3 characters long by adding leading zeros if necessary (e.g., "9" becomes "009").
+// 4. const pounds = paddedPenceNumberString.substring(0, paddedPenceNumberString.length - 2); Extracts the pound portion by taking all but the last two characters (e.g., from "399", this becomes "3").
+// 5. const pence = paddedPenceNumberString.substring(paddedPenceNumberString.length - 2).padEnd(2, "0"); Extracts the last two characters as pence (e.g., from "399", this gives "99"). .padEnd(2, "0") ensures we still get two digits (e.g., from "3" it would become "30").
```
Could we expect this program to work as intended for any valid `penceString` if we deleted `.padEnd(2, "0")` from the code?
In other words, do we really need `.padEnd(2, "0")` in this script?

cjyuan commented on Sprint-1/4-stretch-explore/chrome.md:
```diff
@@ -9,10 +9,14 @@ Let's try an example.
 What effect does calling the `prompt` function have?
+A; This opens a dialog box prompting the user for input.
 What is the return value of `prompt`?
+A; It returns the text the user types in, and stores it in the variable `myName`.
+For example, if I typed "Alice", then `myName` would be "Alice".
```
If we were writing a program that uses `prompt()` to ask for an input value, how could
the program check if the user clicked "OK" or "Cancel"?

cjyuan reviewed (commented):
Explanation is clear. Well done. I only have a few suggestions and questions.

sidicamoli commented:
Updated Sprint-1 after review:

• Improved explanation for count = count + 1 using the correct term.

• Fixed conflict and explained random.js step-by-step with interval notation.

• Refined comments in 3-to-pounds.js and removed .padEnd() as it wasn’t needed.

• Added details in chrome.md about how prompt() works and how to check for cancel/ok.

Thanks to CJ Yuan for the helpful feedback!

cjyuan commented on Sprint-1/4-stretch-explore/chrome.md:
```diff
@@ -20,3 +20,13 @@ A; This opens a dialog box prompting the user for input.
+If the user clicks "Cancel", prompt() returns null.
+
+Example:
+if (myName = null) {
```
Equality operator is  `==`

cjyuan reviewed (commented):
Changes look good.

Leaving responses directly in the comment threads makes it easier to track the discussion. You can try the approach in future PRs.
![image](https://github.com/user-attachments/assets/88193e1a-f6cf-40ec-a04a-7a470f142c6b)
